doc/notes/189_reporting_ant.adoc

```diff
@@ -0,0 +1,224 @@
+statistics of the Protocol Verifier.
+
+==== 3.1.1 Monitored System Reporting
+
+Following is a summary enumeration with details following.
+
+* instantaneous throughput (events per second) (speedometer)
+* throughput (events per second) over a time period ("area under the curve" line graph)
+* jobs succeeded versus failed versus alarmed (pie chart:  green, yellow, red)
+* list of alarmed job names with counters next to them counting the alarms sorted by incidence
+* list of failed job names with counters next to them counting the failures sorted by incidence
+* list of successful job names with counters next to them counting them sorted by incidence
+* list of retried job names with counters next to them counting them sorted by incidence
+* proportion of happy versus unhappy jobs (pie chart:  blue, purple)
+
+===== 3.1.1.1 Instantaneous Throughput
+
+The instantaneous throughput will display the throughput of the
+Protocol Verifier "right now".  It may be time-averaged for smoothing but
+the time window will be short.  This could be displayed in something like
+a speedometer dial.
+
+The units shall be events per second.
+
+===== 3.1.1.2 Throughput Over Time
+
+A graph of the throughput of the Protocol Verifier shall be displayed
+supplying a view of the throughput over the last (say) 24 hours.
+Optionally the display window could be adjustable.  The graph may be
+coloured in an "area under the curve" fashion.
+
+The units shall be events per second.
+
+===== 3.1.1.3 Alarmed versus Failed versus Successful Jobs
+
+A relative proportion of jobs that have ended in alarm conditions, failure
+and success shall be displayed.  A pie chart with the different conditions
+shown as different colours (e.g. red for alarm, yellow for failure and
+green for success) shall make the proportions visually clear.
+
+The units shall be total alarm/failure/success over a noted time period
+(e.g. "today" or "last 24 hours").
+
+===== 3.1.1.4 List of Alarmed Jobs
+
+A list of job types that experienced alarm condition shall be displayed.
+The names of the jobs shall be shown in a readable format.  A counter of
+the number of alarms (over some period of time) shall be shown.  The list
+shall be sorted in descending order starting with the job type receiving
+the greatest incidence of alarms.
+
+===== 3.1.1.5 List of Failed Jobs
+
+A list of job types that failed shall be displayed.  The display format
+shall follow the format described in <<3.1.1.4 List of Alarmed Jobs>>.
+
+===== 3.1.1.6 List of Successful Jobs
+
+A list of job types that succeeded shall be displayed.  The display format
+shall follow the format described in <<3.1.1.4 List of Alarmed Jobs>>.
+
+===== 3.1.1.7 List of Retried Jobs
+
+A list of job types that experienced retries shall be displayed.  The
+names of the jobs shall be shown in a readable format.  A counter of the
+number of retries (over some period of time) shall be shown.  The list
```

Comment:
I'm assuming that if job #1 (of type Job A) is retried, that the retry attempt will be given a unique job id (say #2), and job #2 will include a reference to the job that it is retrying (i.e. #1).
If the retry attempt also fails and the job is retried again, is this a second retry of job #1 or a retry of job #2?
This probably makes no difference to the dashboard requirements but I think it might be important to agree what should be expected.